### PR TITLE
Improve fence specifier handling

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- preserve indentation from orphaned language lines when attaching code fence specifiers
- allow comma separated specifiers with spaces
- document the updated behaviour
- cover the edge cases with tests

## Testing
- `make fmt`
- `make lint`
- `make test`
- `make markdownlint`
- `make nixie` *(fails: too many arguments)*

------
https://chatgpt.com/codex/tasks/task_e_6883f8be90ac8322999b36de77ba4f56

## Summary by Sourcery

Improve code fence specifier handling by preserving indent from orphaned language lines and supporting comma-separated specifiers with spaces, update documentation, and add tests for edge cases

Enhancements:
- Preserve indentation from orphaned specifier lines when attaching code fence specifiers to unindented fences
- Accept and normalize comma-separated specifiers with spaces in code fences

Documentation:
- Document new indentation preservation and space-tolerant comma-separated specifier behavior in function and architecture docs

Tests:
- Add tests for indentation preservation when fences are unindented and for space-tolerant comma-separated specifiers

### CODE REVIEW

File: tests/fences.rs [hunk 1]
```diff
@@ -180,3 +180,17 @@ fn attaches_orphan_specifier_mixed_indent() {
     let out = attach_orphan_specifiers(&compress_fences(&input));
     assert_eq!(out, lines_vec![" \t```rust", " \tfn main() {}", " \t```"]);
 }
+
+#[test]
+fn attaches_orphan_specifier_uses_candidate_indent_when_fence_unindented() {
+    let input = lines_vec!["  Rust", "", "```", "fn main() {}", "```"];
+    let out = attach_orphan_specifiers(&compress_fences(&input));
+    assert_eq!(out, lines_vec!["  ```rust", "fn main() {}", "```"]);
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/fences.rs:188]: Has this now been resolved?

File: src/fences.rs [hunk 2]
```diff
@@ -86,12 +88,31 @@ pub fn attach_orphan_specifiers(lines: &[String]) -> Vec<String> {
                     idx -= 1;
                 }
                 if idx > 0 {
-                    let candidate = out[idx - 1].trim().to_string();
-                    if ORPHAN_LANG_RE.is_match(&candidate)
+                    let candidate_raw = out[idx - 1].as_str();
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/fences.rs]: Has this now been resolved?

File: docs/architecture.md [hunk 3]
```diff
@@ -21,7 +21,9 @@ pub fn process_stream_inner(lines: &[String], opts: Options) -> Vec<String>
 The function combines several helpers documented in `docs/`:
 
 - `fences::compress_fences` and `attach_orphan_specifiers` normalize code block
-  delimiters.
+  delimiters. `attach_orphan_specifiers` keeps indentation from the language
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/architecture.md]: Has this now been resolved?
